velox/benchmarks/basic/LikeBenchmark.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "velox/benchmarks/ExpressionBenchmarkBuilder.h"
+#include "velox/functions/lib/Re2Functions.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+
+using namespace facebook;
+using namespace facebook::velox;
+using namespace facebook::velox::functions;
+using namespace facebook::velox::functions::test;
+using namespace facebook::velox::memory;
+using namespace facebook::velox;
+
+int main(int argc, char** argv) {
+  folly::Init init(&argc, &argv);
+
+  exec::registerStatefulVectorFunction("like", likeSignatures(), makeLike);
+  // Register the scalar functions.
+  prestosql::registerAllScalarFunctions("");
+
+  // exec::register
+  ExpressionBenchmarkBuilder benchmarkBuilder;
+  const vector_size_t vectorSize = 1000;
+  auto vectorMaker = benchmarkBuilder.vectorMaker();
+
+  auto makeInput =
+      [&](vector_size_t vectorSize, bool padAtHead, bool padAtTail) {
+        return vectorMaker.flatVector<std::string>(vectorSize, [&](auto row) {
+          // Strings in even rows contain/start with/end with a_b_c depends on
+          // value of padAtHead && padAtTail.
+          if (row % 2 == 0) {
+            auto padding = std::string("x", row / 2 + 1);
```

Comment:
this should be std::string (n, 'x')

I'll fix. Just FYI.

Reply:
Thanks!

Reply:
@mbasmanova Sorry for this issue, I re-runed the benchmark and update the benchmark results in the PR description, now the speedup is about 10x to 20x, more resonable :)